test/jdk/sun/management/jmxremote/bootstrap/AbstractFilePermissionTest.java

```diff
@@ -141,12 +141,13 @@ private void test1() throws Exception {
         Files.setPosixFilePermissions(file2PermissionTest, perms_0700);
 
         if (doTest() != 0) {
+            System.out.println("FAILURE");
```

Comment:
Would be better to print out which test failed and include the error #.

Reply:
Maybe very marginally useful. I'll add it.  8-)

It's the exit code of the JVM process, being zero or not zero, and the output already printed before the "FAILURE" note that will be the guide as to what went wrong.

If it's the port failure, you'll see the retries then the eventual FAILURE, so that will be clear.

If the JVM failed to startup, it's more the output than the exit code that we want.

I really wavimnted the FAILURE printed because it tests two things, and when you see this test for the first time, it may not be obvious which of the tests is the actual failure - particularly as test2 is expected to cause an error and a non-zero exit code.  So it says "1 failure", and you can stare at two things that look like failures. 8-)

Maybe very marginally useful. I'll add it.  8-)

It's the exit code of the JVM process, being zero or not zero, and the output already printed before the "FAILURE" note will be the guide as to what went wrong.  (If it's the port failure, you'll see the retries then the eventual FAILURE, so that will be clear.)

If the JVM failed to startup, it's more the text output than the exit code that we want.

I really wanted the FAILURE printed because it tests two things, and when you see this test for the first time, it may not be obvious which of the tests is the actual failure - particularly as test2 is expected to cause an error and a non-zero exit code.  So it says "1 failure", and you can stare at two things that look like failures. 8-)